Python/Product/TestAdapter.Executor/TestDiscoverer.cs

```diff
@@ -51,20 +51,19 @@ class ProjectTestDiscover : ITestDiscoverer {
             var sourceToProjSettings = RunSettingsUtil.GetSourceToProjSettings(discoveryContext.RunSettings);
 
             foreach (var testGroup in sources.GroupBy(x => sourceToProjSettings[x])) {
-                DiscoverTestGroup(testGroup, logger, discoverySink);
+                DiscoverTestGroup(testGroup, discoveryContext, logger, discoverySink);
             }
         }
 
-        private void DiscoverTestGroup(IGrouping<PythonProjectSettings, string> testGroup, IMessageLogger logger, ITestCaseDiscoverySink discoverySink) {
+        private void DiscoverTestGroup(IGrouping<PythonProjectSettings, string> testGroup, IDiscoveryContext discoveryContext,  IMessageLogger logger, ITestCaseDiscoverySink discoverySink) {
             PythonProjectSettings settings = testGroup.Key;
-            if (!settings.PytestEnabled) {
-                return;
-            }
 
-            var discovery = new DiscoveryService(logger);
-            var results = discovery.RunDiscovery(settings, testGroup);
-            
-            PyTestDiscoveryReader.ParseDiscovery(results, discoverySink, settings, logger);
+            try {
+                var discovery = DiscovererFactory.GetDiscoverer(settings);
```

Comment:
now we get a discoverer form a factory based on the TestFrameworkType in the projectsettings